pygfx/renderers/wgpu/engine/blender.py

```diff
@@ -91,7 +91,7 @@ def get_color_descriptors(self, blender):
             {
                 "format": blender.pick_format,
                 "blend": None,
-                "write_mask": wgpu.ColorWrite.ALL,
+                "write_mask": wgpu.ColorWrite.ALL if material_write_pick else 0,
             },
```

Comment:
Another option is to drop this whole dict when `material_write_pick` is False. I'm not sure to what extend this is different, but it would probably require applying the same trick to `get_color_attachments()`.

So maybe first see if it works like this (with just setting `write_mask`).

Reply:
I believe I tried this. I got some really strange errors. Looking at it again, I think you are correct that the `get_color_attachments` change is likely also necessary.

<details>

```
$ python examples/feature_demo/image_click_events.py
Detected skylake derivative running on mesa i915. Clears to srgb textures will use manual shader clears.
The fragment stage "fs_main" output @location(1) values are ignored
Draw error
Traceback (most recent call last):
  File "/home/mark/miniforge3/envs/dev/lib/python3.10/site-packages/wgpu/gui/_gui_utils.py", line 77, in log_exception
    yield
  File "/home/mark/miniforge3/envs/dev/lib/python3.10/site-packages/wgpu/gui/base.py", line 135, in _draw_frame_and_present
    self.draw_frame()
  File "/home/mark/git/pygfx/examples/feature_demo/image_click_events.py", line 86, in <lambda>
    canvas.request_draw(lambda: renderer.render(scene, camera))
  File "/home/mark/git/pygfx/pygfx/renderers/wgpu/engine/renderer.py", line 527, in render
    command_buffers += self._render_recording(
  File "/home/mark/git/pygfx/pygfx/renderers/wgpu/engine/renderer.py", line 657, in _render_recording
    render_pass.end()
  File "/home/mark/miniforge3/envs/dev/lib/python3.10/site-packages/wgpu/backends/wgpu_native/_api.py", line 2641, in end
    libf.wgpuRenderPassEncoderEnd(self._internal)
  File "/home/mark/miniforge3/envs/dev/lib/python3.10/site-packages/wgpu/backends/wgpu_native/_helpers.py", line 322, in proxy_func
    raise wgpu_error  # the frame above is more interesting ↑↑
wgpu._classes.GPUValidationError: Validation Error

Caused by:
    In wgpuRenderPassEncoderEnd
      note: encoder = `<CommandBuffer-(0, 1, Vulkan)>`
    In a set_pipeline command
      note: render pipeline = `<RenderPipeline-(1, 1, Vulkan)>`
    Render pipeline targets are incompatible with render pass
    Incompatible color attachments at indices []: the RenderPass uses textures with formats [Some(Rgba8UnormSrgb), Some(Rgba16Uint)] but the RenderPipeline uses attachments with formats [Some(Rgba8UnormSrgb)]
Draw error: Validation Error (2)
Draw error: Validation Error (3)
Draw error: Validation Error (4)
```
</details>

`get_color_attachments` is only used in 1 place where a whole `wobject_list` is passed and not a single object, so I'm worried that changing the attachments isn't really straightforward.
https://github.com/pygfx/pygfx/blob/main/pygfx/renderers/wgpu/engine/renderer.py#L639

Reply:
Oh right ... the color attachment is per-pipeline, the descriptor per object. Then the current solution should be the correct one 👍